src/onnx/parse_matmul.cpp

```diff
@@ -106,6 +108,62 @@ struct parse_matmul : op_parser<parse_matmul>
         return all_zeros;
     }
 
+    static instruction_ref set_scale_arg(const onnx_parser::node_info& info,
+                                         const std::vector<instruction_ref>& args,
+                                         const int index)
+    {
+        instruction_ref scale_arg                            = args[index];
+        std::set<migraphx::shape::type_t> supported_dq_types = {migraphx::shape::float_type,
+                                                                migraphx::shape::half_type};
+
+        if(not(contains(supported_dq_types, scale_arg->get_shape().type())))
+        {
+            MIGRAPHX_THROW("PARSE_QUANT_DOT_SCALDED: Scales must be float or half_type");
```

Comment:
SCALDED?

Reply:
Also, would this message be proper for `MatMul` operator?

Reply:
It won't reach there as its gated by whether the operator contains the scaled inputs. This variant of MatMul also includes the dequantize to convert the quantized input types to float

Reply:
Added naming here instead so errors shared by parsers show the correct parser name